src/main/java/org/wahlzeit_revisited/model/PhotoFactory.java

```diff
@@ -0,0 +1,23 @@
+package org.wahlzeit_revisited.model;
+
+public class PhotoFactory {
```

Comment:
Previously this followed the singleton pattern, right? Not sure if we need a singleton (let's ask Dirk about it), then we should inject an instance, and even better: bind it to an interface rather than the class. This will allow framework users to easily bind their implementation to the interface.

Reply:
Adding an interface is the cleanest way. Binding a `StudentPhotoFactory` to a `PhotoFactory` is possible, but doesn't look that nice

Reply:
My guts feeling is that the situation is "too small" to blow up a single class that is at the same time an interface, an abstract implementation, and a concrete implementation into three different files, the separate interface, abstract superclass, and concrete subclass. Unless, the single class gets too complicated, but I don't think we are there yet. I also like it as an exapmle of the discussion on abstract superclasses in ADAP.